### ISSUE TITLE
[``unused-private-member``] False positives for ``cls`` used in ``__new__`` and in ``__get__``

### ISSUE THREAD

**Issue by stanislavlevin** [PyCQA/pylint#4763]
### Bug description

```python
# pylint: disable=missing-docstring,invalid-name,too-few-public-methods
# pylint: disable=no-value-for-parameter,unused-argument,no-self-use
class A:
    def __new__(cls):
        print(cls.__x())

    @classmethod
    def __x(cls):
        return "__x"

class MyMeta(type):
    def __get__(cls, obj, obj_type):
        print(cls.__x(obj_type))

    def __x(cls, obj_type):
        return "__x"
```

### Command used

```shell
python3 -m pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:8:4: W0238: Unused private member `A.__x(cls)` (unused-private-member)
a.py:15:4: W0238: Unused private member `MyMeta.__x(cls, obj_type)` (unused-private-member)

------------------------------------------------------------------
```

### Expected behavior

There should be no error.

### Pylint version

```shell
pylint 2.9.0+ (including today's main)
```

### OS / Environment

ALTLinux

**Comment by cdce8p** [comment]
This looks like a potential duplicate of #4681

**Comment by Pierre-Sassoulas** [comment]
Closing as it's been fixed at some point before 2.15 or I can't reproduce anymore.
